unlock-app/src/components/interface/checkout/main/Confirm/ConfirmCard.tsx

```diff
@@ -0,0 +1,379 @@
+  const [state, send] = useActor(checkoutService)
+  const config = useConfig()
+  const [isConfirming, setIsConfirming] = useState(false)
+  const {
+    lock,
+    recipients,
+    payment,
+    captcha,
+    paywallConfig,
+    password,
+    promo,
+    metadata,
+  } = state.context
+
+  const { address: lockAddress, network: lockNetwork } = lock!
+
+  const currencyContractAddress = lock?.currencyContractAddress
+
+  const recurringPayment =
+    paywallConfig?.recurringPayments ||
+    paywallConfig?.locks[lockAddress]?.recurringPayments
+
+  const recurringPaymentAmount = recurringPayment
+    ? Math.abs(Math.floor(Number(recurringPayment)))
+    : undefined
+
+  const { mutateAsync: createPurchaseIntent } = usePurchase({
+    lockAddress,
+    network: lockNetwork,
+  })
+
+  const { data: creditCardEnabled } = useCreditCardEnabled({
+    lockAddress,
+    network: lockNetwork,
+  })
+
+  const { mutateAsync: updateUsersMetadata } = useUpdateUsersMetadata()
+
+  const { isInitialLoading: isInitialDataLoading, data: purchaseData } =
+    usePurchaseData({
+      lockAddress: lock!.address,
+      network: lock!.network,
+      promo,
+      password,
+      captcha,
+      paywallConfig,
+      recipients,
+    })
+
+  const {
+    data: pricingData,
+    isInitialLoading: isPricingDataLoading,
+    isError: isPricingDataError,
+  } = usePricing({
+    lockAddress: lock!.address,
+    network: lock!.network,
+    recipients,
+    currencyContractAddress: lock?.currencyContractAddress,
+    data: purchaseData!,
+    paywallConfig,
+    enabled: !isInitialDataLoading,
+    symbol: lockTickerSymbol(
+      lock as Lock,
+      config.networks[lock!.network].nativeCurrency.symbol
+    ),
+  })
+
+  const isPricingDataAvailable =
+    !isPricingDataLoading && !isPricingDataError && !!pricingData
+
+  const amountToConvert = pricingData?.total || 0
+
+  const { data: totalPricing, isInitialLoading: isTotalPricingDataLoading } =
+    useFiatChargePrice({
+      tokenAddress: currencyContractAddress,
+      amount: amountToConvert,
+      network: lock!.network,
+      enabled: isPricingDataAvailable,
+    })
+
+  const { mutateAsync: capturePayment } = useCapturePayment({
+    network: lock!.network,
+    lockAddress: lock!.address,
+    data: purchaseData,
+    referrers: recipients.map((recipient) => getReferrer(recipient)),
+    recipients,
+  })
+
+  const isLoading =
+    isPricingDataLoading || isInitialDataLoading || isTotalPricingDataLoading
+
+  const baseCurrencySymbol = config.networks[lockNetwork].nativeCurrency.symbol
+  const symbol = lockTickerSymbol(lock as Lock, baseCurrencySymbol)
+
+  const onError = (error: any, message?: string) => {
+    console.error(error)
+    switch (error.code) {
+      case -32000:
+      case 4001:
+      case 'ACTION_REJECTED':
+        ToastHelper.error('Transaction rejected.')
+        break
+      default:
+        ToastHelper.error(message || error?.error?.message || error.message)
+    }
+  }
```

Comment:
I think you can move this outside code and use it as default error handler.

Reply:
yeah good point. Let me clean it up.